macchiato.py

```diff
@@ -2,102 +2,168 @@
+    for i in range(indent_levels):
+        prefix = _single_indent * i
+        fake_lines.append(f"{prefix}if True:\n")
 
     # Handle else/elif/except/finally
     try:
-        first_token = next(
+        first_token: Optional[tokenize.TokenInfo] = next(
             tokenize.generate_tokens(iter([first_line.lstrip()]).__next__)
         )
     except tokenize.TokenError:
         first_token = None
     if first_token and first_token.type == tokenize.NAME:
         name = first_token.string
+        prefix = _single_indent * indent_levels
         if name in {"else", "elif"}:
-            lines.insert(n_fake_before, f"{indent * ' '}if True:\n")
-            lines.insert(n_fake_before + 1, f"{indent * ' '}    pass\n")
-            n_fake_before += 2
+            fake_lines.append(f"{prefix}if True:\n")
+            fake_lines.append(f"{prefix}{_single_indent}pass\n")
         elif name in {"except", "finally"}:
-            lines.insert(n_fake_before, f"{indent * ' '}try:\n")
-            lines.insert(n_fake_before + 1, f"{indent * ' '}    pass\n")
-            n_fake_before += 2
-
-    # Detect an unclosed block at the end. Add ‘pass’ at the end of the line if
-    # needed for valid syntax.
-    last_line = lines[-1]
-    n_fake_after = 0
-    if last_line.rstrip().endswith(":"):
-        lines[-1] = last_line.rstrip() + "pass\n"
-        n_fake_after = 1
+            fake_lines.append(f"{prefix}try:\n")
+            fake_lines.append(f"{prefix}{_single_indent}pass\n")
 
-    with tempfile.NamedTemporaryFile(suffix=".py", mode="wt+", delete=False) as fp:
+    return fake_lines
+
+
+def wrap_lines(lines: List[str]) -> Tuple[List[str], WrapInfo]:
+    """Wrap the input lines with fake text, to fake a complete source document."""
+
+    # Strip leading and trailing blank lines.
+    n_blank_before, n_blank_after = count_surrounding_blank_lines(lines)
+    until = len(lines) - n_blank_after
+    lines = lines[n_blank_before:until]
+    if not lines:
+        wrap_info = WrapInfo(n_blank_before=n_blank_before, n_blank_after=n_blank_after)
+        return lines, wrap_info
+
+    # We may need some preceding fake lines and/or a trailing ``pass``.
+    fake_before_lines = _fake_before_lines(lines[0])
+    last_line = lines[-1].rstrip()
+    if last_line.endswith(":"):
+        lines[-1] = last_line + " pass"
+        trailing_fake_pass = True
+    else:
+        trailing_fake_pass = False
 
+    # Return updated text along with info about what was changed.
+    lines = fake_before_lines + lines
+    wrap_info = WrapInfo(
+        n_blank_before=n_blank_before,
+        n_blank_after=n_blank_after,
+        n_fake_before=len(fake_before_lines),
+        trailing_fake_pass=trailing_fake_pass,
+    )
+    return lines, wrap_info
+
+
+def format_lines(lines: List[str], black_args=None) -> List[str]:
+    if black_args is None:
+        black_args = []
+
+    with tempfile.NamedTemporaryFile(suffix=".py", mode="wt+", delete=False) as fp:
```

Comment:
while at it, maybe write the temp file file in the `os.getcwd()`? see my comments in #4.

Reply:
Ok, I've made the change for this and added some tests to see if the config file is detected properly.